### PR TITLE
#269: Update contributing and add pr issue templates

### LINKED ISSUE #269: Would appreciate some more info in CONTRIBUTING.md
When creating the previous snippet I was not sure about a couple of things:

* What is is this after every snippet title (`###`) in the README: `<!-- Example ID: 30f1d3fc-e267-4b30-84ef-4d9e7091ac1a --->`? Should it be added manually or can it be ignored when creating new snippets?
* Where should new snippets be added? Top/bottom of section, doesn't matter?
* What's the difference between the sections (the first two feel very similar)?
* Before the table of contents it says that `markdown-toc -i README.md --maxdepth 3` was used to create it. The pip package `markdown-toc` does not contain either `-i` or `--maxdepth` flags. Which package is meant, or what version of that package? 
* Should the new table of contents entry for the snippet be created with the above command or created manually (in case the above command does more than only add the entry)?

Would be nice if these things could be explained with a sentence or two each in the contributing document. Or if they are explained somewhere already, perhaps link to them in CONTRIBUTING.md? 

Thanks! :)

### PR DESCRIPTION
**Pull request by nifadyev**
Fixes #269 

## Checklist before requesting a review

- [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
- [x] Have you followed the guidelines in [CONTRIBUTING.md](../../CONTRIBUTING.md)?
- [x] Have you performed a self-review?
- [x] Have you added yourself into [CONTRIBUTORS.md](../../CONTRIBUTORS.md)?

Request review from @satwikkansal

### CODE REVIEW

**Comment by nifadyev** [comment]
Hey @satwikkansal . I've decided to prepare templates and update CONTRIBUTING before creating small issues for fixing `markdown` format and migrating to `embedded` snippets. Could you please review and suggest some changes if necessary?

File: .github/ISSUE_TEMPLATE/new_snippet.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+---
+
+## Description
+
+## Snippet preview
+
+## Checklist before requesting a review
+
+* [ ] Have you checked to ensure there aren't other open [Issues](../../../issues) for the same update/change?
+* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
```
Comment (satwikkansal) [inline .github/ISSUE_TEMPLATE/new_snippet.md:15]: Let's rephrase this

Let's add a section, name `Explanation (optional)`

and a question asking Is this snippet similar to any of the existing snippets in the project or issues.

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+<!--- This project only accepts pull requests related to open issues -->
+<!--- If suggesting a new feature or change, please discuss it in an issue first -->
+
+<!--- Auto-close the issue that your PR fixes -->
+Fixes # (issue number)
+
+## Checklist before requesting a review
+
+- [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
```
Comment (satwikkansal) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:13]: this is not needed.

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+- [ ] Have you followed the guidelines in [CONTRIBUTING.md](../../CONTRIBUTING.md)?
+<!--- Question: Should we add pre-commit hook and CI check for commit messages? For example, https://github.com/compilerla/conventional-pre-commit-->
+- [ ] Do your commit messages follow [guidelines](https://www.conventionalcommits.org/en/v1.0.0/)?
```
Comment (satwikkansal) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:16]: this is also not needed.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,8 +1,19 @@
+# Contributing
+
+## Getting Started
+
+Contributions are made to this repo via Issues and Pull Requests (PRs). A few general guidelines that cover both:
+
+- Search for existing Issues and PRs before creating your own.
+- We work hard to makes sure issues are handled in a timely manner but, depending on the impact, it could take a while to investigate the root cause. A friendly ping in the comment thread to the submitter or a contributor can help draw attention if your issue is blocking.
```
Comment (satwikkansal) [inline CONTRIBUTING.md:8]: 👍

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,8 +1,19 @@
...
+
+## Issues
+
 All kinds of patches are welcome. Feel free to even suggest some catchy and funny titles for the existing Examples. The goal is to make this collection as interesting to read as possible. Here are a few ways through which you can contribute,
 
-- If you are interested in translating the project to another language (some people have done that in the past), please feel free to open up an issue, and let me know if you need any kind of help.
+- If you are interested in translating the project to another language, please feel free to open up an issue using `translation` template, and let me know if you need any kind of help.
```
Comment (satwikkansal) [inline CONTRIBUTING.md:14]: Sorry I might have forgotted, Is the `translation` template already merged.

**Comment by satwikkansal** [comment]
> Hey @satwikkansal . I've decided to prepare templates and update CONTRIBUTING before creating small issues for fixing `markdown` format and migrating to `embedded` snippets. Could you please review and suggest some changes if necessary?

@nifadyev just did and left some comments, the overarching theme is that we shouldn't make it time consuming for a new contributor to set up an issue or open a PR, as it demotivates them to share something that might be of interest to us. Once they create an issue we can ask for extra details then, that approach has better conversion :)

File: CONTRIBUTING.md (hunk 5 above)
Comment (nifadyev) [reply CONTRIBUTING.md:14]: No, it stored in my local `git stash` for now, I will add it to current PR a bit later, just waited for your comments about the whole PR

**Comment by nifadyev** [comment]
> we shouldn't make it time consuming for a new contributor to set up an issue or open a PR

100%. The problem is to find the balance between ease of creating issue/PR and avoiding duplicates and extra work for reviewers/maintainers. Let's start as you suggested and then adopt if necessary.
Anyway, a good maintained project should not have a lot of open issues and PRs, so points from checklist are not so important

**Comment by nifadyev** [comment]
Hey @satwikkansal, I've fixed your remarks and added PR and issue templates for bug reports and translations

File: .github/ISSUE_TEMPLATE/new_snippet.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+-->
+## Description
+
+## Snippet preview
+
+## Checklist before calling for maintainers
+
+* [ ] Have you checked to ensure there aren't other open [Issues](../../../issues) for the same update/change?
+* [ ] Have you checked that this snippet is not similar to any of the existing snippets?
+* [ ] Have you added an `Explanation` section? It shall include the reasons of changes and why you'd like us to include them
```
Comment (satwikkansal) [inline .github/ISSUE_TEMPLATE/new_snippet.md:21]: let's mark the explanation section optional, sometime people come to us without having exact explanation

**Comment by satwikkansal** [comment]
@nifadyev, thanks! One minor comment, everything else looks good

**Comment by nifadyev** [comment]
Thanks, @satwikkansal , comment about `Explanation` section has been added

**Comment by satwikkansal** [comment]
Thanks @nifadyev, I merged!

**Pull request merged**